src/database/gravity-db.c

```diff
@@ -1407,3 +1408,42 @@ bool gravityDB_get_regex_client_groups(clientsData* client, const unsigned int n
 
 	return true;
 }
+
+void check_inaccessible_adlists(void){
+
+	// check if any adlist was inaccessible in the last gravity run
+	// if so, gravity stored `status` in the adlist table with
+	// "3": List unavailable, Pi-hole used a local copy
+	// "4": List unavailable, there is no local copy available 
+
+	// Do not proceed when database is not available
+	if(!gravityDB_opened && !gravityDB_open())
+	{
+		logg("check_inaccessible_adlists(): Gravity database not available");
+		return;
+	}
+
+	const char *querystr = "SELECT id, address FROM adlist WHERE status IN (3,4) AND enabled=1";
+	
+	// Prepare query
+	sqlite3_stmt *query_stmt;
+	int rc = sqlite3_prepare_v2(gravity_db, querystr, -1, &query_stmt, NULL);
+	if(rc != SQLITE_OK){
+		logg("check_inaccessible_adlists(): %s - SQL error prepare: %s", querystr, sqlite3_errstr(rc));
+		gravityDB_close();
+		return;
+	}
+
+    // Perform query
+    while((rc = sqlite3_step(query_stmt)) == SQLITE_ROW)
+    {
+        int id = sqlite3_column_int(query_stmt, 0);
+        const char *address = strdup((const char*)sqlite3_column_text(query_stmt, 1));
+
+        // log to the message table
+        logg_inaccessible_adlist(id, address);
+    }
+
+    // Finalize statement
+    sqlite3_finalize(query_stmt);
```

Comment:
This section uses spaces instead of tabs (used anywhere else in this file).

Reply:
I'll change that. Maybe a good point to think about https://github.com/pi-hole/FTL/issues/1368 again.